Sources/SSChart/Bar/Stacked/StackedBarChart.swift

```diff
@@ -103,93 +98,64 @@ extension StackedBarChart {
 extension StackedBarChart {
     func calculateChartData() {
         let totalValue = items.reduce(CGFloat(0)) { currentSum, item in
-            assert(item.value > 0, "[SSChart] StackedBarChartItem should have positive value.")
+            assert(item.value >= 0, "[SSChart] StackedBarChartItem can't have negative value.")
             return currentSum + item.value
         }
         
-        maxBarWidth = bounds.width - (Double((items.count-1)) * itemSpacing)
-        
-        var currentTotalValue: CGFloat = 0
+        var prefixSum: CGFloat = 0
         
         items.forEach { item in
-            percentages.append(ChartItemValuePercentage(start: (currentTotalValue / totalValue), end: (currentTotalValue + item.value) / totalValue))
-            currentTotalValue += item.value
+            percentages.append(ChartItemValuePercentage(start: prefixSum / totalValue, end: (prefixSum + item.value) / totalValue))
+            prefixSum += item.value
         }
     }
 }
 
 // MARK: - draw
 extension StackedBarChart {
     func drawChart() {
-        if items.isEmpty {
-            let bar = createBar(frame: CGRect(x: 0, y: 0, width: bounds.size.width, height: bounds.size.height), color: .systemGray)
-            addSubview(bar)
-            bars.append(bar)
-            
-            if showPercentage {
-                let label = createPercentageLabel(percentage: ChartItemValuePercentage(start: 0, end: 0))
-                label.center = CGPoint(x: bar.bounds.width/2, y: bar.bounds.height/2)
-                bar.addSubview(label)
-            }
-            return
-        }
-        
-        for (index, (item, percentage)) in zip(items, percentages).enumerated() {
-            drawBars(index, item, percentage, showPercentage: showPercentage)
-        }
+        drawBars()
+        maskChart()
     }
     
-    private func drawBars(_ index: Int, _ item: StackedBarChartItem, _ percentage: ChartItemValuePercentage, showPercentage: Bool) {
-        
-        let bar = createBar(frame: CGRect(x: (maxBarWidth * percentage.start) + (Double(index) * itemSpacing), y: barMoveUpSpace, width: maxBarWidth * (percentage.end - percentage.start), height: frame.height), color: item.color)
-        addSubview(bar)
-        bars.append(bar)
-        
-        if showPercentage {
-            let label = createPercentageLabel(percentage: percentage)
-            label.center = CGPoint(x: bar.bounds.width/2, y: bar.bounds.height/2)
-            
-            bar.addSubview(label)
+    func drawBars() {
+        for (item, percentage) in zip(items, percentages) {
+            let pieceOfBarLayer =
+            createBarLayer(with: percentage, color: item.color)
+            barLayer.addSublayer(pieceOfBarLayer)
```

Comment:
Find more about performance difference when using UIView and CALayer for drawing.